test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -309,8 +334,9 @@ public void ParseValue ()
             // TODO: is it possible to get the system to recognize when a complete unicode char is present and start the next one then, without a space?
 //            Assert.Equal("日本", Parsers.Value.End().Parse(@"\xe6\x97\xa5\xe6\x9c\xac"));
 
-            Assert.Throws<ParseException>(() => Parsers.Value.End().Parse("0\n1"));
-            Assert.Throws<ParseException>(() => Parsers.Value.End().Parse("'"));
+            Assert.Equal("0", Parsers.Value.Parse("0\n1").Value);   // value ends on linebreak
```

Comment:
Again, I am very uncomfortable with introducing these behaviors.

Reply:
And again I will definitely have a closer look and add tests and/or explain better ;)

Reply:
Added the test in here too.

I will try to rewrite Unquoted-Matcher to throw on leading quotes, thinking about it this will be much cleaner (if you have to explain your tests so deeply may shows there is some problem in the code I guess :D )

Reply:
Part of the problem with these tests (now that I have refreshed myself on all this code) is that they should never happen. If the larger context sent this kind of text to be parsed by an quoted value (i.e. with a newline in the middle), something has gone very wrong in that larger context, which is the Assignment parser.

Which is to further say that if this did happen, it should throw an error, ideally, because things are definitely broken somewhere. But it also should not happen, and I am disinclined to add tests for this sort of case, and definitely disinclined to change any code to support this case.

Reply:
I adapted some of your code about the comment detection, because Sprache makes it really hard to have a working positive look-ahead. (which in fact is done by the combination of first parsing whitespaces and then throwing if we have a `hash`-character)

And there are quite some tests which enforce throwing exceptions at all the cases where we do not have a valid UnquotedValue-Parsing, even more than before, because before we just relied on the Assignment-Parser to consume all whitespaces before ValueParser, but now it is enforced by tests.